test/e2e/json-rpc/wallet_addEthereumChain.spec.ts

```diff
@@ -0,0 +1,329 @@
+          await openDapp(driver);
+
+          const beforePermittedChains = await getPermittedChains(driver);
+
+          assert.deepEqual(beforePermittedChains, []);
+
+          const switchEthereumChainRequest = JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: '0x53a',
+                chainName: 'Localhost 8546 alternative',
+                nativeCurrency: {
+                  name: '',
+                  symbol: 'ETH',
+                  decimals: 18,
+                },
+                // this does not match what already exists in the NetworkController
+                rpcUrls: ['http://127.0.0.1:8546'],
+                blockExplorerUrls: [],
+              },
+            ],
+          });
+
+          await driver.executeScript(
+            `window.ethereum.request(${switchEthereumChainRequest})`,
+          );
+
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+          await driver.clickElement({ text: 'Approve', tag: 'button' });
+
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+          const afterPermittedChains = await getPermittedChains(driver);
+          assert.deepEqual(afterPermittedChains, ['0x53a']);
+        },
+      );
+    });
+
+    it('prompts to switch to the chain when the rpc endpoint being added already exists', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withNetworkControllerDoubleGanache()
+            .build(),
+          ganacheOptions: {
+            ...defaultGanacheOptions,
+            concurrent: [{ port: 8546, chainId: 1338 }],
+          },
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver }: { driver: Driver }) => {
+          await unlockWallet(driver);
+          await openDapp(driver);
+
+          const beforePermittedChains = await getPermittedChains(driver);
+
+          assert.deepEqual(beforePermittedChains, []);
+
+          const switchEthereumChainRequest = JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: '0x53a',
+                chainName: 'Localhost 8546',
+                nativeCurrency: {
+                  name: '',
+                  symbol: 'ETH',
+                  decimals: 18,
+                },
+                // this matches what already exists in the NetworkController
+                rpcUrls: ['http://localhost:8546'],
+                blockExplorerUrls: [],
+              },
+            ],
+          });
+
+          await driver.executeScript(
+            `window.ethereum.request(${switchEthereumChainRequest})`,
+          );
+
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+          await driver.clickElement({ text: 'Confirm', tag: 'button' });
```

Comment:
Can we make some assertions about what is displayed on this confirmation

Reply:
https://github.com/MetaMask/metamask-extension/pull/29837/commits/d9b397bc50daa893c137db0f3d5af8635af065f7